app/src/main/java/doit/study/droid/quiz/OneTestSummaryFragment.kt

```diff
@@ -0,0 +1,138 @@
+package doit.study.droid.quiz
+
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.animation.PropertyValuesHolder
+import android.animation.TypeEvaluator
+import android.animation.ValueAnimator
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.animation.BounceInterpolator
+import android.view.animation.DecelerateInterpolator
+import android.widget.TextView
+
+import androidx.fragment.app.Fragment
+
+import com.mikhaellopez.circularprogressbar.CircularProgressBar
+import doit.study.droid.R
+import doit.study.droid.views.DeathStarLoader
+import kotlin.math.roundToInt
+
+
+class OneTestSummaryFragment : Fragment() {
+    private val progressBar by lazy { view!!.findViewById<View>(R.id.progressBar) as DeathStarLoader }
+    private val textSummary by lazy { view!!.findViewById<View>(R.id.textSummary) as TextView }
+    private val wrongCnt by lazy { view!!.findViewById<View>(R.id.wrong_cnt) as TextView }
+    private val rightCnt by lazy { view!!.findViewById<View>(R.id.right_cnt) as TextView }
+    private val percentageOfRightAnswers by lazy {
+        arguments!!.let {
+            it.getInt(RIGHT_CNT_KEY) * 100 / (it.getInt(RIGHT_CNT_KEY) + it.getInt(WRONG_CNT_KEY))
+        }
+    }
+    private var animatorSet: AnimatorSet? = null
+
+
+    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_result_test, parent, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        showTextSummary()
+        arguments!!.let {
```

Comment:
let was used for logical grouping.
The motivation for dangerous bang-bang to fail earlier, because the fragment is not functional without proper arguments.

Do you suggest to check arguments in onCreate with an explicit exception?